core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -889,18 +889,23 @@ public void listStatus(AlluxioURI path, ListStatusContext context,
    * descendantType. The result is returned via a list specified by statusList, in postorder
    * traversal order.
    *
+   * @param context call context
    * @param rpcContext the context for the RPC call
    * @param currInodePath the inode path to find the status
    * @param auditContext the audit context to return any access exceptions
    * @param descendantType if the currInodePath is a directory, how many levels of its descendant
-   *                       should be returned
+   *        should be returned
    * @param resultStream the stream to receive individual results
    * @param depth internal use field for tracking depth relative to root item
    */
-  private void listStatusInternal(RpcContext rpcContext, LockedInodePath currInodePath,
-      AuditContext auditContext, DescendantType descendantType, ResultStream<FileInfo> resultStream,
-      int depth) throws FileDoesNotExistException, UnavailableException, AccessControlException,
-      InvalidPathException {
+  private void listStatusInternal(ListStatusContext context, RpcContext rpcContext,
+      LockedInodePath currInodePath, AuditContext auditContext, DescendantType descendantType,
+      ResultStream<FileInfo> resultStream, int depth) throws FileDoesNotExistException,
+      UnavailableException, AccessControlException, InvalidPathException {
+    // Fail if the client has cancelled the rpc.
+    if (context.isCancelled()) {
+      throw new RuntimeException("Call cancelled by the client.");
```

Comment:
Is `RuntimeException` the correct way to "cancel" an rpc thread? Also, for whichever exception we throw, can we include more in the message, like the type of call, and the path?

Reply:
What do you mean by correct? For the propagation, `RPCUtils` is prepared for `RuntimeException`s. Alternatively I'd define a new AlluxioStatusException type for cancelled calls and throw that instead.

Reply:
For adding more info to the messages, please check `RpcUtils`. It'll take care of putting all the details of the call.

Reply:
I see, it logs a lot of useful things. I guess the actual exception message doesn't have to be descriptive, since the client will never get this information (since the client canceled and is gone)